### ISSUE TITLE
Enable/Disable Workload Automation

### ISSUE THREAD

**Issue by stevenpall** [flux-web/flux-web#29]
This would be a useful feature for rollback scenarios. If the workload is automated and a different image is selected, the change will simply get rolled forward again. How much work would be involved in getting this functionality added?

**Comment by stevenpall** [comment]
@idobry @ybaruchel Just wondering if this is something you'd be willing to work on in the near term? Thanks!

**Comment by derkinderfietsen** [comment]
Hi @stevenpall, can you explain it in detail? I am not pretty sure I understood the request.

**Comment by stevenpall** [comment]
@derkinderfietsen The requested feature would add functionality to disable and re-enable release automation. This is already something that can be achieved via a change to the corresponding `HelmRelease` (`fluxcd.io/automated: "true"`) or via `fluxctl` (https://docs.fluxcd.io/en/1.18.0/references/fluxctl.html#turning-off-automation).

The advantage of having it be a part of flux-web is that developers and other non-privileged users could then disable a release, rollback to a previous version, and re-enable the release once a fixed version is pushed.

**Comment by AndreasPrang** [comment]
Hi @derkinderfietsen,

we would like to contribute this feature. Please have a look on our approach: 

- When rolling back to a specific version automation will be disabled
- An option will be added on top of the menu of the entries in "Available Tags" as shown in the link below to enable automation
- latest tag/image version will be visible in the menu two times. 
-- the first to anable automation
-- the second to pin the latest version / disable automation

Please have a look at our scribble: https://moveoffice-my.sharepoint.com/:i:/g/personal/andreas_prang_axelspringer_com/ESp_u_UAE4BMl_Y1XA3B_WMB-PuSqixMVt600_9s_6-3Og?e=wI6hAy

**Comment by derkinderfietsen** [comment]
@AndreasPrang 
Thanks for the detailed explanation, please feel free to submit a PR with this feature request.

Please note that the automation/deautomation API works in two ways:

**Automation**
A POST API call must be made against `/api/flux/v9/update-manifests` with the following payload:
```json
{
  "type": "policy",
  "cause": {
    "Message": "",
    "User": "John Doe <john@example.com>"
  },
  "spec": {
    "default:deployment/helloworld": {
      "add": {
        "automated": "true"
      },
      "remove": {}
    }
  }
}
```

**Deautomation**
A POST API call must be made against the same URL `/api/flux/v9/update-manifests` but with a different payload:
```json
{
  "type": "policy",
  "cause": {
    "Message": "",
    "User": "John Doe <john@example.com>"
  },
  "spec": {
    "default:deployment/helloworld": {
      "add": {},
      "remove": { // Note the difference here
        "automated": "true"
      }
    }
  }
}
```

Then, we must make short-polled calls to `/api/flux/v6/jobs?id=`**Job ID From Update API Call** for getting the status of the current job and then updating the frontend about the final result (success/error) through the websocket, please note that this logic is already implemented in workload releases.

**Comment by AndreasPrang** [comment]
Hi @derkinderfietsen,

we are very happy to provide the PR on this feature. 


Best regards,
Andreas

**Comment by AndreasPrang** [comment]
Hi @derkinderfietsen,

do you have an update for us regarding the open PR?

**Comment by derkinderfietsen** [comment]
Hi @AndreasPrang, thanks for the PR, very good solution.

Merged (https://github.com/flux-web/flux-web/pull/113) and Landed in [v0.3](https://github.com/flux-web/flux-web/releases/tag/v0.3)
